2025/lab/Keys/Week5_categorical_and_nonlinear_predictors.Rmd

```diff
@@ -0,0 +1,505 @@
+
+Let's try this now, turning the variables `sex`, `race`, `partyid`, and `degree` into a series of dummies.
+
+```{r dummy-create, warning=FALSE}
+
+gss <- gss %>% 
+  fastDummies::dummy_cols(select_columns = c("sex", "race", "partyid", "degree"))
+
+names(gss)
+
+```
+
+
+Now, let's try out running models with our new dummies. Note, too, that it doesn't matter which dummy we leave out as long as one of them is left out. Changing the dummy will only change the intercept, not the slope. 
+```{r ols-dummy-test}
+ols_nodummy <- lm(income ~ age + prestige + workhrs + race + sex,
+                  data = gss)
+ols_dummy1 <- lm(income ~ age + prestige + workhrs + 
+                   race_White + race_Other +
+                   sex_Female,
+                 data = gss)
+ols_dummy2 <- lm(income ~ age + prestige + workhrs +  
+                   race_White + race_Other +
+                   sex_Male,
+                   data = gss)
+
+modelsummary(list(ols_nodummy, ols_dummy1, ols_dummy2),
+             stars = T, title = "Testing Dummy Variables",
+             gof_omit = "F|Log|IC|RMSE")
+
+```
+
+As we can see from these results, there is no difference in whether you use your own dummies or use the defaults. Just note, however, that R will drop whatever the first category is. In the case of `sex`, the first one was `male`, since I coded it as a `factor`. In the case of `race`, however, it was simply a `character`-type variable, so the first category, and the one that was dropped, was `black`. Creating and using your own dummies gives you more control over this, but isn't necessary. 
+
+If we'd like to change the "reference category," AKA the category that is dropped, one option is to turn the variable into a factor with new levels (1 = White, 2 = Black, 3 = Other). Another is by putting in ourselves which levels to include in the model: Including Black and Other makes White the reference. Finally, another option is to use the `stats::relevel()` function to choose a new base category. (`relevel(var, ref = new_reference_level)`)
+
+# Exercise: Specifiy a multivarite model excluding white, males, and less than a high school degree. Interpret the intercept.
+```{r exercise 2}
+
+names(gss)
+
+ols_dummy3 <- lm(income ~ age + prestige + workhrs + 
+                   race_Black + race_Other +
+                   sex_Female + 
+                   degree_HS + degree_Assoc + degree_Bach + degree_Grad,
+                 data = gss)
+
+modelsummary(ols_dummy3,
+             estimate = "{estimate}{stars}",
+             fmt = 0,
+             statistic = NULL,
+             gof_omit = "IC|Log|alg|pss|F|RMSE",
+             notes = "Notes: + p < 0.1, * p < 0.05, ** p < 0.01, *** p < 0.001",
+             output="huxtable")
+
+```
+
+```{r lets make it a picture}
```

Comment:
@brenthoagland Not a huge deal, but I recommend using snake_case and avoiding leaving any words in a standard sentence in code. There are other cases where there are spaces in chunk names.